corehq/apps/registration/utils.py

```diff
@@ -313,3 +321,89 @@ def send_mobile_experience_reminder(recipient, full_name, additional_email_conte
         logging.warning(
             "Can't send email, but the message was:\n%s" % message_plaintext)
         raise
+
+
+@run_only_when(not settings.ENTERPRISE_MODE and not settings.UNIT_TESTING)
+@silence_and_report_error("Exception raised in the check username availability rate limiter",
+                          'commcare.registration.check_username_rate_limiter_errors')
+def rate_limit_check_username_availability():
+    """
+    This holds attempts per IP OR per session below limits given by
+    check_username_rate_limiter_by_ip and check_username_rate_limiter_by_session, respectively.
+
+    Requests without both an IP and session ID are rejected.
+    """
+    _status_session_rate_limited = 'session_rate_limited'
+    _status_ip_rate_limited = 'ip_rate_limited'
+    _status_bad_request = 'bad_request'
+    _status_accepted = 'accepted'
+
+    def get_session_and_ip():
+        request = get_request()
+        if request and request.session:
+            return request.session.session_key, get_ip(request)
+        else:
+            return None
```

Comment:
should return `None, None`

Reply:
https://github.com/dimagi/commcare-hq/pull/35688/commits/b48eb3154e26d569573d39cdd49cf3beb870de96